docs/src/dev_guide.md

```diff
@@ -0,0 +1,53 @@
+# Development Guide
+
+In this file we explain the release and developing processes we follow in all of the QED-project Julia packages.
+
+## General
+
+We use the [GitFlow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) workflow. In short, this means we use a `dev` branch to continuously develop on, and a `main` branch that has the latest released version. Once enough changes have been merged into dev to warrant a new release, a release branch is created from `dev`, adding the latest changelog and increasing the version number. It is then merged into `main` and also back into `dev` to create the new common root of the two branches.
+
+In the following, the different possible situations are explained in more detail.
+
+## Merging a Feature Branch
+
+To create a new feature, create a new branch from the current `dev` and push it onto a fork. Once you have developed, committed, and pushed the changes to the repository, open a pull request to `dev` of the official repository. Automatic tests will then check that the code is properly formatted, that the documentation builds, and that tests pass. If all of these succeed, the feature branch can be merged after approval of one of the maintainers of the repository.
+
+### CI Testing
+
+There are two types of tests that run automatically when a pull request is opened: Unit tests, which only test the functionality in the current repository, and integration tests, which check whether other downstream packages (i.e. packages that depend on the current repository) still work with the code updates.
+
+In case breaking changes are introduced by the pull request the integration tests will fail and an additional pull request has to be opened in each breaking package, fixing the incompatibility. For details on how this works, refer to the [CI documentation](ci.md#Integration-Tests-for-CI-Users).
+
+## Releasing a New Version
+
+Several steps are involved in releasing a new version of a package. For versioning, we follow [Julia's versioning guidelines](https://julialang.org/blog/2019/08/release-process/). However, we do not provide patches for previous minor or major versions.
+
+### Release Issue Template
+
+The following is a template that can be used to open issues for a specific release of one of the QED-project packages. Customize it by simply replacing `<version>` with the version to-be-released.
+
+```md
```

Comment:
I rendered the documentation locally. It does not looks like expected. Some of the text format characters are ignored and some of characters are correct rendered but the character are still visible.

For example `*After*` is rendered italic, but you still see the asterisk.

Reply:
Yes, but that is intended, it's only the  preview and if it wasn't like that, copying it also wouldn't copy the asterisks, \` characters and so on.